cmake.d/FrameworkFunctions.cmake

```diff
@@ -0,0 +1,63 @@
+#
+# Backport of new target_sources that uses relative paths
+#
+function(quicc_target_sources target)
```

Comment:
do we even need `quicc_target_sources`?

Reply:
I opened issue #23  for this. I think it can be removed but it is still used in many places. This should be done as a separate PR.